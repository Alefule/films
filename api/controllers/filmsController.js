const Film = require('../models/Film');
const films = require('../../data/json/films.json');

exports.showFilms = (req,res) => res.json(films);

exports.showOneFilm = (req,res) => {
    let id = req.params.id;
    let film = films.find(film => film.id == id);
    res.send(film);
}

exports.addFilm = (req, res) => {
    let film = {...req.body}
    let filmToAdd = new Film(film.id,film.name);
    films.push(filmToAdd);
    res.json(films);
}

exports.deleteFilm = (req, res) => {
    let idToDelete = req.params.id;
    for (let i = 0; i < films.length; i++) {
        if (films[i].id == idToDelete) {
            films.splice(i,1);
        }
    }
    res.json(films);
}

exports.updateFilm = (req, res) =>{
    let idToUpadte = req.params.id;
    let filmToUpdate = {...req.body};
    for (let i = 0; i < films.length; i++) {
        if(films[i].id == idToUpadte){
            films[i].id = films[i].id;
            films[i].name = filmToUpdate.name || films[i].name;
            films[i].likes = filmToUpdate.likes || films[i].likes;
        }
    }
}



