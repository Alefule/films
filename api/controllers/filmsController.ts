const Film = require('../models/Film');
const films = require('../../data/json/films.json');

export const showFilms = () => films.map(movie => {
    const newMovie = {...movie};
    delete newMovie.likes;
    delete newMovie.password;
    return newMovie;
});

exports.showOneFilm = (req, res) => {
    let id = req.params.id;
    let film = films.find(film => film.id == id);
    res.send(film);
};

exports.addFilm = (req, res) => {
    let film = { ...req.body };
    let filmToAdd = new Film(film.id, film.name);
    films.push(filmToAdd);
    res.json(films);
};

exports.deleteFilm = (req, res) => {
    let idToDelete = req.params.id;
    for (let i = 0; i < films.length; i++) {
        if (films[i].id == idToDelete) {
            films.splice(i, 1);
        }
    }
    res.json(films);
};

exports.updateFilm = (req, res) => {
    let idToUpdate = req.params.id;
    let filmToUpdate = { ...req.body };
    for (let i = 0; i < films.length; i++) {
        if (films[i].id == idToUpdate) {
            films[i].name = filmToUpdate.name || films[i].name;
            films[i].likes = filmToUpdate.likes || films[i].likes;
        }
    }
};



