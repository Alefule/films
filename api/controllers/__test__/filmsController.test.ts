import { showFilms } from "../filmsController";

const checkJsonFields = (movie) => {
    if (Object.keys(movie)[1] == 'name'
        && Object.keys(movie)[0] == 'id') {

        return true;
    }
    console.log(Object.keys(movie));
    return false;
};

describe('filmsController', () => {
    describe('showFilms()', () => {
        test('should return movies ids and names', () => {
            const movies = showFilms();
            // console.log(movies);
            // expect(movies.every(checkJsonFields)).toBeTruthy();
            // console.log(movies);
            expect(movies).toMatchSnapshot();

        });
    });
});
