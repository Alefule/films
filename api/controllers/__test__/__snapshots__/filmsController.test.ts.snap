// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filmsController showFilms() should return movies ids and names 1`] = `
Array [
  Object {
    "id": "1",
    "name": "Titanic",
  },
  Object {
    "id": "2",
    "name": "Batman",
    "password": "5678",
  },
  Object {
    "id": "3",
    "name": "Piratas del Caribe",
    "password": "1234",
  },
  Object {
    "id": "4",
    "name": "Harry Potter",
  },
  Object {
    "id": "5",
    "name": "Pretty Woman",
  },
]
`;
